stdlib/LinearAlgebra/src/triangular.jl

```diff
@@ -1917,68 +1914,224 @@ function log(A0::UpperTriangular{T}) where T<:BlasFloat
     end
 
     # Compute the Padé approximation
-    Y = zeros(T, n, n)
+    t = eltype(A)
+    Y = zeros(t, n, n)
     for k = 1:m
-        Y = Y + w[k] * (A / (x[k] * A + I))
+        Y = Y + t(w[k]) * rdiv_quasitriu(A, t(x[k]) * A + I)
```

Comment:
This looks like it could be allocating quite a bit. It might be possible to make the `rdiv` in-place` and then also update `Y` in-place.

Reply:
We absolutely could cut down on allocations here by using a temporary to store `x[k] * A + I`, modify it in-place for the solve, and then use `axpby!` to update `Y` inplace. The matrix function implementations mostly don't try to reduce allocations for some reason, and I had left it this way for similarity to the old implementation.

Reply:
Yeah. I can see that it's mostly the sea as before. I was just trying to see if I could find some of the extra allocations that visible in your timings but it probably can't be this line that is causing them.